gpflow/base.py

```diff
@@ -192,55 +202,55 @@ def assign(
         )
 
     @property
-    def is_tensor_like(self):
+    def is_tensor_like(self) -> UNKNOWN:
         """
         This method means that TensorFlow's `tensor_util.is_tensor` function
         will return `True`
         """
         return True
 
     @property
-    def name(self):
+    def name(self) -> str:
         return self._unconstrained.name
 
     @property
-    def initializer(self):
+    def initializer(self) -> UNKNOWN:
         return self._unconstrained.initializer
 
     @property
-    def device(self):
+    def device(self) -> Optional[str]:
         return self._unconstrained.device
 
     @property
-    def dtype(self):
+    def dtype(self) -> tf.DType:
         return self._unconstrained.dtype
 
     @property
-    def op(self):
+    def op(self) -> tf.Operation:
         return self._unconstrained.op
 
     @property
-    def shape(self):
+    def shape(self) -> tf.TensorShape:
         if self.transform is not None:
             return self.transform.forward_event_shape(self._unconstrained.shape)
         return self._unconstrained.shape
 
-    def numpy(self):
+    def numpy(self) -> np.ndarray:
         return self.read_value().numpy()
 
-    def get_shape(self):
+    def get_shape(self) -> tf.TensorShape:
         return self.shape
 
-    def _should_act_as_resource_variable(self):
+    def _should_act_as_resource_variable(self) -> UNKNOWN:
```

Comment:
It's not unknown, it's simply not returning anything.
```suggestion
    def _should_act_as_resource_variable(self):
```
Blame it on tensorflow internals

Reply:
I've tried to type all these methods to be as they are on Tensor and Variable, but I couldn't find anything on this method

what do you mean by "not returning anything"?

Reply:
btw by `UNKNOWN` I mean types that I just can't figure out, not that the types are inherently unknown. It's so that everything has an explicit type, which makes it much easier to find missing type hints

Reply:
alternatively, I could add comments to each explaining we don't know the type

Reply:
`->None`?

Reply:
agreed, that's the type as it is here, but as mentioned above, I've tried not to type things based on how they're implemented here. For example, I'd expect `_should_act_as_resource_variable` to actually return `bool` given the name, but I don't really know

Reply:
> alternatively, I could add comments to each explaining we don't know the type, and leave off the type hint

yes please, thanks